orchestra/static/orchestra/project_management/js/data_service.js

```diff
@@ -1,40 +1,62 @@
 (function() {
-  'use strict';
+  // 'use strict';
 
   var serviceModule = angular.module('orchestra.project_management');
 
-  serviceModule.factory('dataService', function($rootScope, $location, orchestraApi) {
+  serviceModule.factory('dataService', function($location, $rootScope, $route, orchestraApi) {
     /**
      * Service to share and manipulate project management data across
      * visualization components.
      */
     var _meta;
     var _now = new Date();
 
-    return {
-      setup: function(projectId) {
+    var service = {
+      currentProject: {short_description: 'Select project to display', id: null},
+      dataReady: null,
+      allProjects: {},
+      resetData: function(projectId) {
+        ds = this;
         _meta = {
           'tasks': {}
         };
-        this.projectId = projectId;
       },
-      updateData: function(cb) {
+      getAllProjects: function() {
+        var dataService = this;
+        dataService.dataReady = orchestraApi.allProjects().then(function(response) {
+          response.data.forEach(function(project) {
+            dataService.allProjects[project.id] = project;
+          });
+        });
+        return dataService.dataReady;
+      },
+      changeProject: function(projectId) {
```

Comment:
any chance you somehow have a request out to modify a project via projman, and then change the project, which results in unexpected behavior?